lib/access-context.js

```diff
@@ -143,19 +144,9 @@ AccessContext.prototype.getUserId = function() {
     // or permit to resolve a valid user model
     var userModel = this.registry.findModel(p.type);
     if (!userModel) continue;
-    if (userModel.modelName === 'User') {
+    if (userModel.prototype instanceof BaseUser) {
       return p.id;
     }
```

Comment:
> Just one last check: you're confident in model prototypal inheritance in loopback and there's no way a model extending 'User' can return false to `instanceof BaseUser`?

Yes, I am pretty confident:

 - Model inheritance in LoopBack uses Node.js' `util.inherits` under the hood, see [model-builder.js#L200](https://github.com/strongloop/loopback-datasource-juggler/blob/ad9934159b3f8b9c7600c9942b07bd6d1a9460d3/lib/model-builder.js#L200)

 - We are already relying on `instanceof` in the following places:

   - [app.model()](https://github.com/strongloop/loopback/blob/a34f321d2b54f2a2b16d3cd63c1adce8273d8cf3/lib/application.js#L129)
   - [registry.getModelByType()](https://github.com/strongloop/loopback/blob/c3ab9320127cb7726e83b9668f641f7cb2a55692/lib/registry.js#L340)
   - [Model.nestRemoting](https://github.com/strongloop/loopback/blob/b86eac5cf6030f9df32b180d049eef37a04d3a4b/lib/model.js#L940)